src/overlays/actors/ovl_Arms_Hook/z_arms_hook.c

```diff
@@ -23,27 +29,329 @@ const ActorInit Arms_Hook_InitVars = {
+                hookedActor = NULL;
+                this->hookedActor = NULL;
+            } else {
+                if (this->actor.attachedB != 0) {
+                    sp94 = func_8002DB48(this, hookedActor);
+                    sp90 = sqrtf(SQ(this->hookedActorDistDiff.x) + SQ(this->hookedActorDistDiff.y) +
+                                 SQ(this->hookedActorDistDiff.z));
+                    Math_Vec3f_Diff(&hookedActor->posRot.pos, &this->hookedActorDistDiff, curPos);
+                    if (50.0f < (sp94 - sp90)) {
+                        ArmsHook_DetachHookFromActor(this);
+                        hookedActor = NULL;
+                    }
+                }
+            }
+        }
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_Arms_Hook/func_808650AC.s")
+        hookBodyDistDiff =
+            Math_Vec3f_DistXYZAndStoreDiff(&player->unk_3C8, &this->actor.posRot.pos, &hookBodyDistDiffVec);
+        if (hookBodyDistDiff < 30.0f) {
+            velocity = 0.0f;
+            phi_f16 = 0.0f;
+        } else {
+            if (this->actor.attachedB != NULL) {
+                velocity = 30.0f;
+            } else {
+                if (hookedActor != NULL) {
+                    velocity = 50.0f;
+                } else {
+                    velocity = 200.0f;
+                }
+            }
+            phi_f16 = hookBodyDistDiff - velocity;
+            if (hookBodyDistDiff <= velocity) {
+                phi_f16 = 0.0f;
+            }
+            velocity = phi_f16 / hookBodyDistDiff;
+        }
+        newPos.x = hookBodyDistDiffVec.x * velocity;
+        newPos.y = hookBodyDistDiffVec.y * velocity;
+        newPos.z = hookBodyDistDiffVec.z * velocity;
+        hookBodyDistDiffVecPtr = &hookBodyDistDiffVec;
+        if (this->actor.attachedB == NULL) {
+            if ((hookedActor != NULL) && (hookedActor->id == ACTOR_BG_SPOT06_OBJECTS)) {
+                Math_Vec3f_Diff(&hookedActor->posRot.pos, &this->hookedActorDistDiff, curPos);
+                phi_f16 = 1.0f;
+            } else {
+                Math_Vec3f_Sum(&player->unk_3C8, &newPos, curPos);
+                if (hookedActor != NULL) {
+                    Math_Vec3f_Sum(curPos, &this->hookedActorDistDiff, &hookedActor->posRot.pos);
+                }
+            }
+        } else {
+            Math_Vec3f_Diff(hookBodyDistDiffVecPtr, &newPos, &player->actor.velocity);
+            player->actor.posRot.rot.x =
+                atan2s(sqrtf(SQ(hookBodyDistDiffVecPtr->x) + SQ(hookBodyDistDiffVecPtr->z)), -hookBodyDistDiffVec.y);
+        }
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_Arms_Hook/func_80865148.s")
+        if (phi_f16 < 50.0f) {
+            ArmsHook_DetachHookFromActor(this);
+            if (phi_f16 == 0.0f) {
+                ArmsHook_SetupAction(this, &func_80864FC4);
+                if (ArmsHook_AttachToPlayer(this, player) != 0) {
```

Comment:
Im wondering about this function name. It definitely seems to be attaching to the player, however the return value seems to depend on hooks attachedB which doesnt seem related to the player attachment at all. I could be wrong though. 
Either way, since this seems to be a boolean return you can get rid of != 0

Reply:
lmao I messed up the preview, I mean ArmsHook_AttachToPlayer